pkg/osinfo/layer_scanner.go

```diff
@@ -117,12 +116,26 @@ func (loss *LayerScanner) extractFileFromTar(tarPath, filePath, destPath string)
 	}
 	defer f.Close()
 
+	// Read the first bytes to determine if the file is compressed
+	var sample [2]byte
+	var compressed bool
+	if _, err := io.ReadFull(f, sample[:]); err != nil {
+		return errors.Wrap(err, "sampling bytes from tar header")
+	}
+	if _, err := f.Seek(0, 0); err != nil {
+		return errors.Wrap(err, "rewinding read pointer")
+	}
+
+	if sample[0] == 31 && sample[1] == 139 {
```

Comment:
- Do you want to use the hex versions (easier to spot, like the ones https://github.com/golang/go/blob/master/src/compress/gzip/gunzip.go#L20-L22 )
- Do you want to sample 3 bytes? (looks like the 3rd byte is `0x08` for deflate)

Reply:
Nice suggestion, i've added it and added a link in a comment to the gunzip module for reference.